examples/scene_tree.rs

```diff
@@ -0,0 +1,171 @@
+use i_cant_believe_its_not_bsn::*;
+
+use bevy::color::palettes::*;
+use bevy::prelude::*;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_plugins(scene_tree_plugin)
+        .run();
+}
+
+/// Plugin for the editor scene tree pane.
+fn scene_tree_plugin(app: &mut App) {
+    app.add_systems(Startup, mock_setup_pane);
+    app.add_systems(Update, build_scene_tree);
+}
+
+const DEFAULT_BACKGROUND: Srgba = tailwind::NEUTRAL_600;
+const SELECTED_BACKGROUND: Srgba = tailwind::NEUTRAL_700;
+const HOVERED_BACKGROUND: Srgba = tailwind::NEUTRAL_500;
+
+// Mock an editor pane.
+fn mock_setup_pane(mut commands: Commands) {
+    commands.insert_resource(Selection { entity: None });
+
+    commands.spawn((Name::new("Camera"), Camera2d, Hover { count: 0 }));
+    commands.spawn((Name::new("Example Entity"), Hover { count: 0 }));
+
+    commands.spawn_empty().build(template! {
+        // A full-screen node which centers it's concent
+        Node {
+            width: Val::Percent(100.0),
+            height: Val::Percent(100.0),
+            align_items: AlignItems::Center,
+            justify_content: JustifyContent::Center,
+            ..default()
+        } => [
+            // The actual window we will use as the root of the scene tree
+            (
+                SceneTreeRoot,
+                Node {
+                    flex_direction: FlexDirection::Column,
+                    column_gap: Val::Px(2.0),
+                    padding: UiRect::all(Val::Px(8.0)),
+                    ..default()
+                },
+                BackgroundColor(DEFAULT_BACKGROUND.into()),
+            );
+        ];
+    });
+
+    // Add observers to drive input interaction.
+    commands.add_observer(select_row);
+    commands.add_observer(hover_over_row);
+    commands.add_observer(hover_leave_row);
+}
+
+// Build the scene tree row template.
+fn build_scene_tree(
+    mut commands: Commands,
+    scene_tree: Single<Entity, With<SceneTreeRoot>>,
+    query: Query<(Entity, &Name, Option<&Hover>)>,
+    selection: Res<Selection>,
+) {
+    let template: Template = query
+        .iter()
+        .map(|(entity, name, hover)| scene_tree_row(entity, name, hover, &selection))
+        .flatten()
+        .collect();
+
+    commands.entity(*scene_tree).build_children(template);
```

Comment:
Notice how this is building a template on a child of another template. Now that receipts are stored on-entity, you can apply any template to any entity and it will *just work* (remove any components/children included in the previous template but not the new one).